src/modules/phoneNumber/formatPhoneNumber.ts

```diff
@@ -0,0 +1,49 @@
+import { withErrorBoundary } from '../../common/errorBoundary';
+import { PHONE_FORMATTER_MAPPER } from './data/phoneFormatterMapper';
+import { detectCountryCodeFromDialCode, cleanPhoneNumber } from './utils';
+
+const formatPhoneNumber = (
+  phoneNumber: string | number,
+  countryCode?: keyof typeof PHONE_FORMATTER_MAPPER,
+): string => {
+  if (!countryCode) countryCode = detectCountryCodeFromDialCode(phoneNumber);
```

Comment:
Priority will still be `detectCountryCodeFromDialCode` since we are keeping `countryCode` as an optional parameter in case a user wants to override default. 


By default, libphonenumber-js uses the country code passed as a parameter to format the phone number. However, if the country code is not explicitly provided, it attempts to detect it from the phone number itself and then formats it accordingly.